mathesar/api/ui/permissions/ui_records.py

```diff
@@ -0,0 +1,65 @@
+from rest_access_policy import AccessPolicy
+
+from mathesar.api.utils import get_table_or_404
+from mathesar.models.users import DatabaseRole, Role, SchemaRole
+
+
+class RecordAccessPolicy(AccessPolicy):
+    """
+    Anyone can view a Record as long as they are a superuser or have
+    at least a Viewer access to the Schema or the Database of the Table.
+    The permissions trickle down, so if someone has a Viewer Access for a Database
+    They automatically become a Schema Viewer
+    Refer https://wiki.mathesar.org/en/product/specs/users-permissions#database-permissions
+    Only superuser or schema/database Manager/Editor can delete/modify/create a Record
+    """
+    statements = [
+        {
+            'action': ['list', 'retrieve'],
```

Comment:
There is no need for these statements as the UI namespace `RecordViewset` contains only the `delete` endpoint. So please remove the irrelevant statements.